shell/platform/darwin/macos/framework/Source/FlutterAccessibilityMac.mm

```diff
@@ -0,0 +1,575 @@
+NSString* const NSAccessibilityTextChangeValueLength = @"AXTextChangeValueLength";
+NSString* const NSAccessibilityTextChangeValues = @"AXTextChangeValues";
+
+enum AXTextStateChangeType {
+  AXTextStateChangeTypeUnknown,
+  AXTextStateChangeTypeEdit,
+  AXTextStateChangeTypeSelectionMove,
+  AXTextStateChangeTypeSelectionExtend
+};
+
+enum AXTextEditType {
+  AXTextEditTypeUnknown,
+  AXTextEditTypeDelete,
+  AXTextEditTypeInsert,
+  AXTextEditTypeTyping,
+  AXTextEditTypeDictation,
+  AXTextEditTypeCut,
+  AXTextEditTypePaste,
+  AXTextEditTypeAttributesChange
+};
+
+const int kLiveRegionChangeIntervalMS = 20;
+
+extern "C" {
+
+// The following are private accessibility APIs required for cursor navigation
+// and text selection. VoiceOver started relying on them in Mac OS X 10.11.
+
+AXTextMarkerRef AXTextMarkerCreate(CFAllocatorRef allocator, const UInt8* bytes, CFIndex length);
+
+}  // extern "C"
+
+// Static.
+FlutterAccessibility* FlutterAccessibility::Create() {
+  return new FlutterAccessibilityMac();
+}
+
+FlutterAccessibilityMac::FlutterAccessibilityMac() {
+  ax_platform_node_ = AXPlatformNode::Create(this);
+  FML_DCHECK(ax_platform_node_);
+}
+
+FlutterAccessibilityMac::~FlutterAccessibilityMac() {
+  ax_platform_node_->Destroy();
+}
+
+void FlutterAccessibilityMac::Init(AccessibilityBridge* bridge, AXNode* node) {
+  FlutterAccessibility::Init(bridge, node);
+  std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> convert;
+  old_text_editing_value_ = convert.from_bytes(GetAXNode()->GetValueForControl());
+}
+
+void FlutterAccessibilityMac::OnAccessibilityEvent(AXEventGenerator::TargetedEvent targeted_event) {
+  FML_DCHECK(targeted_event.node == GetAXNode());
+  ax::AXEventGenerator::Event event_type = targeted_event.event_params.event;
+  gfx::NativeViewAccessible native_node = GetNativeViewAccessible();
+  FML_DCHECK(native_node);
+
+  NSString* mac_notification = nil;
+
+  switch (event_type) {
+    case ax::AXEventGenerator::Event::ACTIVE_DESCENDANT_CHANGED:
+      if (GetData().role == ax::Role::kTree) {
+        mac_notification = NSAccessibilitySelectedRowsChangedNotification;
+      } else if (GetData().role == ax::Role::kTextFieldWithComboBox) {
+        // Even though the selected item in the combo box has changed, we don't
+        // want to post a focus change because this will take the focus out of
+        // the combo box where the user might be typing.
+        mac_notification = NSAccessibilitySelectedChildrenChangedNotification;
+      } else {
+        // In all other cases we should post
+        // |NSAccessibilityFocusedUIElementChangedNotification|, but this is
+        // handled elsewhere.
+        return;
+      }
+      break;
+    case ax::AXEventGenerator::Event::LOAD_COMPLETE:
+      // On MacOS 10.15, firing AXLoadComplete causes focus to move to the
+      // webpage and read content, despite the "Automatically speak the webpage"
```

Comment:
This looks like it's copied from Chromium source - can we update it to make it clearer about what this means for a Flutter application?